Recruit-Training/Advanced-Recruit-Training/Embedded-Teaser/Training/Core/Src/Training.c

```diff
@@ -2,5 +2,59 @@
 
 void trainingTask(uint8_t* data)
 {
+	validData = 0;
+	
+	if (motorCheck(*data) && motorCheck(*data) == motorCheck(*(data + 1)) && lightsCheck(*(data + 2)) ) // set outputArray if motors data are valid, motors are in sync and lights data is valid
```

Comment:
I would say it's a better idea to store the return values of these functions in variables. That way you don't need to call the functions multiple times since you call them again in the set of if statements at the bottom of the function